daemon/k8s/init.go

```diff
@@ -154,12 +129,11 @@ func WaitForNodeInformation(ctx context.Context, k8sGetter k8sGetter) error {
 		}).Info("Received own node information from API server")
 
 		useNodeCIDR(n)
-
-		restoreRouterHostIPs(n)
+		restoreRouterHostIPs(n, log)
 	} else {
 		// if node resource could not be received, fail if
 		// PodCIDR requirement has been requested
-		if option.Config.K8sRequireIPv4PodCIDR || option.Config.K8sRequireIPv6PodCIDR {
+		if requireIPv4CIDR || requireIPv6CIDR {
 			log.Fatal("Unable to derive PodCIDR via Node or CiliumNode resource, giving up")
```

Comment:
I understand this line is not part of this PR but seems a bit odd to fatal here while the function returns an error? Might be better to propagate an error and let the call site decide how to best handle it?

Reply:
Changed. 
Both callers to this function should handle returned error properly.